lib/tools/ts-inspector.js

```diff
@@ -79,6 +83,95 @@ var parsePsi_ = function(bytes, pmt) {
   }
 };
 
+var nextPesPacket_ = function(bytes, pmt, startIndex, type) {
+  var
+    packet,
+    endIndex = startIndex + MP2T_PACKET_LENGTH;
+
+  while (endIndex < bytes.byteLength) {
+    // Look for a pair of start and end sync bytes in the data..
+    if (bytes[startIndex] === SYNC_BYTE && bytes[endIndex] === SYNC_BYTE) {
+      // We found a packet
+      packet = bytes.subarray(startIndex, endIndex);
+
+      if (probe.ts.parseType(packet, pmt.pid) === 'pes' &&
+          probe.ts.parsePesType(packet, pmt.table) === type) {
+        return {
+          packet: packet,
+          startIndex: startIndex,
+          endIndex: endIndex
+        };
+      }
+
+      startIndex += MP2T_PACKET_LENGTH;
+      endIndex += MP2T_PACKET_LENGTH;
+      continue;
+    }
+
+    // If we get here, we have somehow become de-synchronized and we need to step
+    // forward one byte at a time until we find a pair of sync bytes that denote
+    // a packet
+    startIndex++;
+    endIndex++;
+  }
+
+  return null;
+};
+
+var getFullAdtsHeader_ = function(bytes, pmt, packet, startIndex, endIndex) {
+  var
+    headerOffset,
+    headerBytes,
+    i,
+    crcPresent,
+    headerLength,
+    nextPacketResult,
+    packetsAvailable = (bytes.byteLength - startIndex - 1) / MP2T_PACKET_LENGTH;
+
+  for (i = 0; i < packetsAvailable; i++) {
+    headerOffset = probe.ts.getAdtsHeaderOffset(packet);
+
+    if (headerOffset) {
```

Comment:
I think this has the potential to break for headers split across pes packets. From what I can tell, if the header is split across packets, the syncword will only be in the first packet (or split itself), so the second half of the packet may or may not get appended depending on if any of the values in the packet satisfy the sync word check.